src/bp/core/services/telemetry/sdk-methods.ts

```diff
@@ -0,0 +1,185 @@
+    }
+  }
+
+  private async SDKUsage(): Promise<SDKUsageEvent> {
+    const bots = await this.botService.getBotsIds()
+    return { actions: await this.getActionUsages(bots), hooks: await this.getHooksUsages() }
+  }
+
+  private async getActionUsages(bots: string[]): Promise<SDKUsage> {
+    const rootFolder = 'actions'
+    const globalActionsNames = await this.ghostService
+      .global()
+      .directoryListing('/', `${rootFolder}/*.js`)
+      .map(path => path.split('/').pop() || '')
+
+    const global = await Promise.map(globalActionsNames, this.parseFile(`/${rootFolder}`))
+    const perBots = await Promise.reduce(bots, this.botActionsReducer(rootFolder), [] as ParsedFile[])
+
+    return { global, perBots }
+  }
+
+  private botActionsReducer(rootFolder: string): (acc: any, botId: any) => Promise<ParsedFile[]> {
+    return async (acc, botId) => {
+      const botActionsNames = await this.ghostService.forBot(botId).directoryListing(`/${rootFolder}`, '*.js')
+      const parsedFiles = await Promise.map(botActionsNames, this.parseFile(`/${rootFolder}`, { botId }))
+      return [...acc, ...parsedFiles]
+    }
+  }
+
+  private async getHooksUsages(): Promise<SDKUsage> {
+    const hooksPayload = await getHooksLifecycle(this.botService, this.ghostService)
+    const global = await this.parseHooks(hooksPayload.global.filter(hook => hook.type === 'custom'))
+    const perBots = await Promise.reduce(hooksPayload.perBots, this.botHooksReducer(), [] as ParsedFile[])
+
+    return { global, perBots }
+  }
+
+  private botHooksReducer(): (acc: any, botHooks: BotHooks) => Promise<ParsedFile[]> {
+    return async (acc, botHooks: BotHooks) => {
+      const { botId, hooks } = botHooks
+      const parsedFiles = await this.parseHooks(
+        hooks.filter(hook => hook.type === 'custom'),
+        botId
+      )
+      return [...acc, ...parsedFiles]
+    }
+  }
+
+  private async parseHooks(hooks: Hook[], botId?: string): Promise<ParsedFile[]> {
+    return Promise.map(hooks, async hook => {
+      const { lifecycle, enabled, path } = hook
+      const usageParams = { lifecycle, enabled }
+
+      if (botId) {
+        _.set(usageParams, 'botId', botId)
+      }
+
+      return this.parseFile('/hooks', usageParams)(path!)
+    })
+  }
+
+  private parseFile(rootFolder: string, usageParams?): (name: string) => Promise<ParsedFile> {
+    return async (name: string) => {
+      const file = await this.readFileAsString(rootFolder, name, _.get(usageParams, 'botId'))
+      const functions = this.extractFunctions(parse(file, PARSE_CONFIG))
+      const usage = { fileName: name.split('/').pop(), usages: this.parseFunctions(functions) }
+      return { ...usage, ...usageParams }
+    }
+  }
+
+  private parseFunctions(functions: string[]): Usage[] {
+    const sdkFunctions = _.countBy(functions.filter(method => method.split('.')[0] === 'bp'))
+    return _.map(sdkFunctions, (count, fn) => {
+      const [, module, functionName] = fn.split('.')
```

Comment:
are we sure all function calls will always be included in a namespace?

Is something like this possible ? `bp.someFunction()`

Or better, what about : `bp.firstLevelNamespace.secondLevelNamespace.someFunction()`

when splitting `fn.split('.')` we kind of know that the last element is the actual function call and the first element is bp. The rest in the middle should be the module.

don't you agree?

Reply:
Agreed, I will use the last element as the function and the middle as the module

Reply:
fixed : a91a805ec44c0ee89679e630addfbd96b7ca0aea